### PR TITLE
MOTECH-3023: Fixed 500 Error during first run

### PR DESCRIPTION
**Pull request by mkruszynski**

### CODE REVIEW

**Comment by kmadej** [comment]
I suppose the problem is with datanucleus. I think that the best solution will be detached user object in findByUserName method

**Comment by mkruszynski** [comment]
@kmadej I checked a solution with detached user object. The error was still occuring. This error exists only when user's roles are null. I checked and for user without any role, the read roles are an empty list, so null is some kind of incorrect read.

File: platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java [hunk 1]
```diff
@@ -30,7 +30,17 @@
...
+            return null;
+        }
+
+        MotechUser retrievedUser = dataService.findByUserName(userName.toLowerCase());
+
+        if (retrievedUser != null && retrievedUser.getRoles() == null) {
+            return findByUserName(userName);
+        }
+
+        return retrievedUser;
```
Comment (kmadej) [inline platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java]: it is not the best solution. Try use getDetachedField to retrieve roles object

File: platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java [hunk 2]
```diff
@@ -30,7 +30,17 @@
      */
     @Transactional
     public MotechUser findByUserName(String userName) {
-        return null == userName ? null : dataService.findByUserName(userName.toLowerCase());
+        if (userName == null) {
+            return null;
```
Comment (kmadej) [inline platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java]: please use only one return statement

File: platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java [hunk 3]
```diff
@@ -30,7 +30,17 @@
...
+        }
+
+        MotechUser retrievedUser = dataService.findByUserName(userName.toLowerCase());
+
+        if (retrievedUser != null && retrievedUser.getRoles() == null) {
```
Comment (kmadej) [inline platform/web-security/src/main/java/org/motechproject/security/repository/MotechUsersDao.java]: please add comment to this code and provide ticket number on JIRA

**Pull request merged**